app/src/main/java/fr/free/nrw/commons/MediaDataExtractor.java

```diff
@@ -34,23 +42,81 @@ public MediaDataExtractor(MediaWikiApi mwApi,
 
     /**
      * Simplified method to extract all details required to show media details.
-     * It fetches media object, deletion status and talk page for the filename
+     * It fetches media object, deletion status and talk page and caption for the filename
      * @param filename for which the details are to be fetched
      * @return full Media object with all details including deletion status and talk page
      */
     public Single<Media> fetchMediaDetails(String filename) {
         Single<Media> mediaSingle = getMediaFromFileName(filename);
         Single<Boolean> pageExistsSingle = mediaClient.checkPageExistsUsingTitle("Commons:Deletion_requests/" + filename);
         Single<String> discussionSingle = getDiscussion(filename);
-        return Single.zip(mediaSingle, pageExistsSingle, discussionSingle, (media, deletionStatus, discussion) -> {
+        Single<String> captionSingle = getCaption(filename);
+        Single<JsonObject> depictionSingle = getDepictions(filename);
+        return Single.zip(mediaSingle, pageExistsSingle, discussionSingle, captionSingle, depictionSingle, (media, deletionStatus, discussion, caption, depiction) -> {
             media.setDiscussion(discussion);
+            media.setCaption(caption);
+            media.setDepiction(formatDepictions(depiction));
             if (deletionStatus) {
                 media.setRequestedDeletion();
             }
             return media;
         });
     }
 
+    /**
+     * Obtains captions using filename
+     * @param filename
+     *
+     * @return caption for the image in user's locale
+     */
+
+    private Single<String> getCaption(String filename) {
+        return mediaClient.getCaptionByFilename(filename);
+    }
+
+    /**
+     * From the Json Object extract depictions into an array list
+     * @param mediaResponse
+     * @return List containing map for depictions, the map has two keys,
+     *  first key is for the label and second is for the url of the item
+     */
+
+    private ArrayList<Map<String, String>> formatDepictions(JsonObject mediaResponse) {
+        try {
+            JsonArray depictionArray = (JsonArray) mediaResponse.get("Depiction");
+            ArrayList<Map<String, String>> depictedItemList = new ArrayList<>();
+            try {
+                for (int i = 0; i <depictionArray.size() ; i++) {
+                    JsonObject depictedItem = (JsonObject) depictionArray.get(i);
+                    Map <String, String> depictedObject = new HashMap<>();
+                    depictedObject.put("label", depictedItem.get("label").toString().substring(3, depictedItem.get("label").toString().length()-3));
```

Comment:
Please add comments explaining why the operation, in particular the `3`, and `1` below.

Reply:
done